### ISSUE TITLE
Follow likers function can't fetch likers properly

### ISSUE THREAD

**Issue by flackoon** [InstaPy/InstaPy#3573]
Getting likers from user:  targetuser1

shuffling links
Got  24 , returning  1  links:  ['https://www.instagram.com/p/{postId}/']
Few likes, not guaranteed you don't follow these likers already.
Got photo likers:  ['', 'username12315', '3,742 others']

Earlier today everything was alright. I run it now and it's this way.
@timgrossmann

**Comment by GraduErd** [comment]
same issue.

**Comment by JoeFertig** [comment]
same here:
_______________________________________
Few likes, not guaranteed you don't follow these likers already.
Got photo likers:  ['', 'bobbyjt88', '1,661 others']  

INFO [2018-12-14 08:39:10] [X]  Failed to get followers count of 'b'1,661 others''  ~empty list
INFO [2018-12-14 08:40:01] [X]  Failed to get following count of 'b'1,661 others''  ~empty list
INFO [2018-12-14 08:40:01] [X]  User: '1,661 others'  |> followers: unknown  |> following: unknown  |> relationship ratio: unknown
ERROR [2018-12-14 08:40:04] [X]  ~cannot get number of posts for username
INFO [2018-12-14 08:40:04] [X]  --> Not a valid user: ---> Sorry, couldn't check for number of posts of username

INFO [2018-12-14 08:41:07] [X]  Failed to get followers count of 'b'''  ~empty list
INFO [2018-12-14 08:42:02] [X]  Failed to get following count of 'b'''  ~empty list
INFO [2018-12-14 08:42:02] [X]  User: ''  |> followers: unknown  |> following: unknown  |> relationship ratio: unknown
ERROR [2018-12-14 08:42:13] [X]  ~cannot get number of posts for username
INFO [2018-12-14 08:42:13] [X]  --> Not a valid user: ---> Sorry, couldn't check for number of posts of username

INFO [2018-12-14 08:42:13] [X]  ---> bobbyjt88 has already been followed 1 times

Few likes, not guaranteed you don't follow these likers already.
Got photo likers:  ['', 'this_lad', '1,487 others']  
_________________________________________________________

How to solve this?

**Comment by michaellafrasse** [comment]
Same :(

**Comment by XavierRaul** [comment]
Hello. It's been two days since I have this problem. can someone help me ?

INFO [2018-12-14 12:53:43] [xavierledo_coach] Starting to follow likers..
""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""

Getting likers from user: marinlle

shuffling links
Got 24 , returning 6 links: ['https://www.instagram.com/p/BrUfzqyFHH5/', 'https://www.instagram.com/p/BrIgCxjFgK-/', 'https://www.instagram.com/p/BqjiY5hlTV_/', 'https://www.instagram.com/p/BqnR7ABF0q3/', 'https://www.instagram.com/p/Bqu2ZNiF9L1/', 'https://www.instagram.com/p/Bq9YHAzlCLe/']
Few likes, not guaranteed you don't follow these likers already.
Got photo likers: ['cyrilcoach', '17,208 others']

INFO [2018-12-14 12:55:00] [xavierledo_coach] Failed to get followers count of 'b'17,208 others'' ~empty list

**Comment by s-h-a-d-o-w-pixel** [comment]
Same here

**Comment by yunggoose** [comment]
Same in here, hopefully someone can solve this problem.

**Comment by hddeus** [comment]
someone ?

**Comment by hddeus** [comment]
@opexa

**Comment by hddeus** [comment]
@erdemnohutcu

**Comment by hddeus** [comment]
@JoeFertig

**Comment by hddeus** [comment]
@michaellafrasse

**Comment by hddeus** [comment]
@XavierRaul

**Comment by hddeus** [comment]
@theJacobDev

**Comment by hddeus** [comment]
@timgrossmann

**Comment by spikeyhair** [comment]
Yes, I notice this issue as well.

**Comment by uluQulu** [comment]
Hi @opexa 
I saw your problem while reading @timgrossmann's thread.

It is a problem out of the existing design off a commit - f31dc3f1ef3b79df670426113474dbb32aba6406 from @pez5001 introducing the usage of the `re` module in there (_15 days ago_).
```python
if re.match(r'\d+ others', liked_this[-1].text):
```
That line fails when a button's text is `13,742 others`.

Why?
Cos `\d+` matches the decimals [0-9] but not with a comma.

I have normalized it and **shared a solution** at my live PR- #3460.
To apply it manually, see 49fd6e7c57691c25e92bd8ad819f5c0ff162f05a commit.

If you find it hard to apply, just open up **commenters_util.py** file and,
**replace**,
```python
if re.match(r'\d+ others', liked_this[-1].text): 
```
**with**
```python
if " others" in liked_this[-1].text: 
```
that's it.

>Other changes in that commit are good but not crucial.

---

Cheers 😁
---

**Comment by GraduErd** [comment]
Traceback (most recent call last):
  File "followerd.py", line 3, in <module>
    from instapy import InstaPy
  File "/Users/erdemnohutcu/InstaPy/instapy/__init__.py", line 3, in <module>
    from .instapy import InstaPy
  File "/Users/erdemnohutcu/InstaPy/instapy/instapy.py", line 60, in <module>
    from .commenters_util import extract_information
  File "/Users/erdemnohutcu/InstaPy/instapy/commenters_util.py", line 67
    print ("loading more comments.")
    ^
IndentationError: unexpected indent

After this i got this error

**Comment by uluQulu** [comment]
@erdemnohutcu 
Python has strict rules on indentation. Make sure you have it set properly.
If you can't solve that indentation issue, upload that file here [in a **zip**], I will help, hopefully.

**Comment by spikeyhair** [comment]
I have tested it and it works. Thanks for the fix.

**Comment by hddeus** [comment]
Thanks @uluQulu , you are amazing

**Comment by unipython** [comment]
Hey there I changed the crucial parts but still get the same error. 

```
> ('\nGetting likers from user: ', 'bbcearth', '\n')
> ('Got ', 12, ', returning ', 3, ' links: ', ...)
> Video has no likes?
> Moving on..
```

 I haven't found 

`if re.match(r'\d+ others', liked_this[-1].text): `

in my commenter_util.py, so these are my current lines:

`def likers_from_photo(browser, amount=20):                                       

    user_liked_list = []
    try:
        liked_this = browser.find_elements_by_xpath("//div/article/div[2]/section[2]/div/a")
        likers = []
        for liker in liked_this:
            if "like this" not in liker.text:
                likers.append(liker.text)        
        if check_exists_by_xpath(browser, "//div/article/div[2]/section[2]/div/a"):
            if " others" in liked_this[-1].text:
                element_to_click = liked_this[-1]
            
            elif " likes" in liked_this[0].text:
                element_to_click = liked_this[0]

        else:
            print ("Video has no likes?")
            print ("Moving on..")
            return []
            
        sleep(1)
        click_element(browser, liked_this[0])
        print ("opening likes")
        # update server calls
        #update_activity()    `

Any ideas?

**Comment by uluQulu** [comment]
Hi @unipython 
>I haven't found
`if re.match(r'\d+ others', liked_this[-1].text):`

You have not found it cos you have a quite old copy of **InstaPy** cos that line above was introduced 15 days before I suggested a change on it.
That 15 days ago, there were other fixes, too, which targeted some problems with video medias and you probably face those problems.

If you just `pull upstream master`, I think you will not have any problems.
And yes, the fix I wrote at this thread is now _merged_ and you don't need to apply it manually.

---

Cheers 😁
---

<sub>Check for updates before issuing a problem</sub>
